### PR TITLE
Update from holiday coding

### PR DESCRIPTION
**Pull request by TAJD**
@marinlauber here are the updates to the code base plus additional work to add unit/integration tests plus workflow to run testing.

### CODE REVIEW

**Review by MG-ng — CHANGES_REQUESTED** [review]
I'm not sure how the setup should work now… The README.md still explains the old setup with `conda env create -f environment.yml` 
https://github.com/marinlauber/Python-VPP/pull/14/commits/c7195b313618975e6597bf1861817bdad430f172#diff-9efd195f4e9bfb79ccd456a1d8370fafcc4bcb0b00ea3799222667d2ae818533

**Review by MG-ng — APPROVED** [review]
The streamlit app is very cool! 
Thank you very much!

**Comment by TAJD** [comment]
> I'm not sure how the setup should work now… The README.md still explains the old setup with `conda env create -f environment.yml` Maybe `pip install -r requirements.txt` should be there instead. [c7195b3#diff-9efd195f4e9bfb79ccd456a1d8370fafcc4bcb0b00ea3799222667d2ae818533](https://github.com/marinlauber/Python-VPP/commit/c7195b313618975e6597bf1861817bdad430f172#diff-9efd195f4e9bfb79ccd456a1d8370fafcc4bcb0b00ea3799222667d2ae818533)

Yes, good point. This is because I'm trying to support the streamlit app as well as the existing development method.

I have an idea about refactoring the vpp logic into a standalone package which will help resolve this problem but that's definitely one for wider discussion and a different PR. For now I'm happy to patch issues with the streamlit app until we come up with a better longer term plan.

File: scripts/test_nlopt.py
[the remark this answers is not in the document]
Comment (TAJD) [reply scripts/test_nlopt.py:1]: Strictly speaking this looks like it's just showing that `nlopt` works which is already covered by the api tests. What is the behaviour you want to capture here?

**Comment by TAJD** [comment]
@marinlauber apologies this is a big PR - please can you review it 🙏

**Comment by marinlauber** [comment]
I will try to do it this weekend, thanks!

File: README.md [hunk 1]
```diff
@@ -12,12 +12,12 @@
...
 4. Add all the windage contributions (mast, crew, rigging, etc.)
 5. Optional Delft hydro model
-6. Add dagerboards to the possible appendages  
+6. Add dagger boards to the possible appendages  
 6. ~~tidy plotting and results~~
 
 ### Prerequisites
 
-There are only a few prerequisites to run this code, most python instalation will have them. We advise to use the `environment.yml` file provided in this repo to set-up a new conda environment to use the code (this keeps you machine nice and tidy).
+There are only a few prerequisites to run this code, most python installations will have them. We advise to use the `environment.yml` file provided in this repo to set-up a new conda environment to use the code (this keeps you machine nice and tidy).
```
Comment (marinlauber) [inline README.md:20]: Do we still have the `environment.yml` or is it gone? If so can you change to the `requirements.txt`?

**Review by marinlauber — CHANGES_REQUESTED** [review]
Very good job! Just the `environment.yml`->`requirements.txt` in the `README.md` to be fixed and then I will merge. Cheers, Tom!

File: README.md (hunk 1 above)
Comment (TAJD) [reply README.md]: I think I deleted it - have updated to reflect requirements plus I made a few other changes to it

**Review by marinlauber — APPROVED** [review]

**Pull request merged**